src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,253 @@
+            occupancy_sensor_type_bitmap_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorTypeBitmap)
+            asserts.assert_less_equal(occupancy_sensor_type_bitmap_dut, 0b00000111, "OccupancySensorTypeBitmap attribute is not in valid range")
+        else:
+            logging.info("OccupancySensorTypeBitmap attribute is a mandatory attribute. Test step fails")
+            asserts.fail("Missing mandatory attribute OccupancySensorTypeBitmap")
+
+        self.step(5)
+        if attributes.HoldTimeLimits.attribute_id in attribute_list:
+            if attributes.HoldTime.attribute_id in attribute_list: # check HoldTime conformance
+                hold_time_limits_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTimeLimits)
+
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMin, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMin, 0, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMax, 0xFFFE, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMax, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+            else:
+                logging.info("HoldTime conformance failed.  Test step skipped")
+                asserts.fail("HoldTime conformance is incorrect")
+
+        else:
+            logging.info("HoldTimeLimits not supported. Test step skipped")
+            self.mark_current_step_skipped()
+        
+        self.step(6)
+        if attributes.HoldTime.attribute_id in attribute_list:
+            hold_time_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+
+            asserts.assert_less_equal(hold_time_dut, 0xFFFE, "HoldTime attribute is out of range")
+            asserts.assert_greater_equal(hold_time_dut, 0, "HoldTime attribute is out of range")
+        else:
+            logging.info("HoldTime not supported. The rest of legacy attribute test can be skipped")
+            self.skip_all_remaining_steps(7)
+            return
+
+        self.step(7)
+        if attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list:
+            if ((occupancy_sensor_type_bitmap_dut == 0b00000001) | ((occupancy_sensor_type_bitmap_dut != 0b00000001)&(occupancy_sensor_type_bitmap_dut != 0b00000010)&(occupancy_sensor_type_bitmap_dut != 0b00000100))):

```

Comment:
The test plan says "if PIR sensor, read" so we seem to want to just have a `&` to check if this is a PIR sensor and if yes, do the read and validation.

The test plan does not say "validate that this attribute only exists for PIR sensors" (even if that is what seems reasonable). If we do this extra check outside the test plan, we should document it as comments.

Reply:
This attribute is valid for PIR or when none of the sensor is PIR/Ultrasonic/PhysicalContact.  Will update to clarify the test plan document.